### ISSUE TITLE
Shows duplicate media entry on activity page while user uploads media as comment from lightbox  

### ISSUE THREAD

**Issue by kjnanda** [rtMediaWP/rtMedia#1399]
Please check this visual - https://drive.google.com/file/d/1uNkVbV0FsAZiHpe-uZWZgh3HpC0CHgqg/view

----

```
Test Environment 

1. WP V5.2.2
2.BuddyPress V4.3.0
3. rtMedia core V4.5.7 
```

**Comment by kjnanda** [comment]
Since the cause of the issue is inconsistent and there are many patches which needs to be cleared. it would take time to fix the issue and testing it completely. 

Moving it to next release milestone.

**Comment by surajkrsingh** [comment]
PR:-https://github.com/rtMediaWP/rtMedia/pull/1415

First of all describing the issues for the solution (PR) is made.

Issue description :-
1. Media duplication on comment activity.
2. Create duplicate activity when we comment on private media.
3. Privacy level is not working for comment activity.

**Note :- All cases are generating when buddypress_mediaCommentActivity is off from rtmedia settings.**

After the fix you can test with below testcases:-
1. Check by comment on media activity with all privacy level like private, public, friend and logged-in.
2. Comment on media activity from Lightbox, Single comment media page and Activity comment on same page.
3. Check both above point one by one with below rtmedia settings.
       1.Keep `buddypress_mediaCommentActivity` is `off`
       2.Keep `buddypress_mediaCommentActivity` is `on`

You can check with more testcases if anyone missed here.

**Note :- At the same time you can check the activity privacy on different level like (with friend, logged-in, publicly, privately).**

**Comment by kjnanda** [comment]
This looks good to me. Please merge the PR @surajkrsingh

**Comment by kjnanda** [comment]
This is working as expected and tested with `dev` branch. ✅

**Comment by kjnanda** [comment]
Closing this issue as this has been fixed and release in - V 4.5.8
